Source/JavaScriptCore/dfg/DFGClobberize.h

```diff
@@ -804,15 +806,18 @@ void clobberize(Graph& graph, Node* node, const ReadFunctor& read, const WriteFu
     case ValueMod:
     case ValuePow:
     case ValueBitLShift:
-    case ValueBitRShift:
+    case ValueBitRShift: {
+        bool isLICMPhase = phase == ClobberPhase::LICM;
+        bool hasSideEffect = node->op() == ValueMod || node->op() == ValueDiv || node->op() == ValuePow;
         // FIXME: this use of single-argument isBinaryUseKind would prevent us from specializing (for example) for a HeapBigInt left-operand and a BigInt32 right-operand.
-        if (node->isBinaryUseKind(AnyBigIntUse) || node->isBinaryUseKind(BigInt32Use) || node->isBinaryUseKind(HeapBigIntUse)) {
+        bool isBigIntUse = node->isBinaryUseKind(AnyBigIntUse) || node->isBinaryUseKind(BigInt32Use) || node->isBinaryUseKind(HeapBigIntUse);
+        if (isBigIntUse && !(isLICMPhase && hasSideEffect)) {
             def(PureValue(node));
```

Comment:
We should just say that BigInt always has a side effect for now.
The way to do it is

`read(World)` `write(SideState)` + the above `def` thing.
So,

```
if (node->isBinaryUseKind(AnyBigIntUse) || node->isBinaryUseKind(BigInt32Use) || node->isBinaryUseKind(HeapBigIntUse)) {
    read(World);
    write(SideState);
    def(PureValue(node));
    return;
}
```